src/System.Net.Http/src/Internal/Mail/MailBnfHelper.cs

```diff
@@ -330,9 +333,21 @@ internal static string GetDateTimeString(DateTime value, StringBuilder builder)
                 localBuilder.Append(' ');
             }
 
-            string[] offsetFields = offset.Split(':');
-            localBuilder.Append(offsetFields[0]);
-            localBuilder.Append(offsetFields[1]);
+            int colonIndex = offset.IndexOf(':');
+            if (colonIndex == -1)
+            {
+                throw new InvalidOperationException("TimeSpan.ToString() should contain at least one colon.");
+            }
+
+            int nextColonIndex = offset.IndexOf(':', colonIndex + 1);
+            if (nextColonIndex == -1)
+            {
+                // Set it past the end of the string
+                nextColonIndex = offset.Length;
```

Comment:
We should not be changing behavior by throwing a new exceptions. We should only use `Assert` to verify our own logic errors in programming. So, it sounds like `Assert` isn't necessarily what we should add, but then we shouldn't add new exception types to throw. So, net-net, we shouldn't make the product code change that results in new exceptions being thrown. That would be a behavior change.